### PR TITLE
OpenAI and Fireworks vision services

### PR DESCRIPTION
**Pull request by chadbailey59**
I might try to add Fireworks image gen support tomorrow, but we can also merge this as-is if we need to cut a release

### CODE REVIEW

File: dot-env.template [hunk 1]
```diff
@@ -30,3 +30,6 @@ PLAY_HT_API_KEY=...
 
 # OpenAI
 OPENAI_API_KEY=...
+
+# Fireworks
+FIREWORKS_API_KEY=...
```
Comment (aconchillo) [inline dot-env.template:35]: Rebase, this should be in already.

File: examples/foundational/03-still-frame.py [hunk 2]
```diff
@@ -35,7 +35,8 @@ async def main(room_url):
                 image_size="square_hd"
             ),
             aiohttp_session=session,
-            key=os.getenv("FAL_KEY"),
+            key_id=os.getenv("FAL_KEY_ID"),
+            key_secret=os.getenv("FAL_KEY_SECRET"),
```
Comment (aconchillo) [inline examples/foundational/03-still-frame.py:39]: No, the new way with the new `fal-client` is using `FAL_KEY`

File: src/dailyai/services/ai_services.py [hunk 3]
```diff
@@ -109,12 +109,20 @@ def __init__(self, **kwargs):
 
     @abstractmethod
     async def run_vision(self, frame: VisionImageFrame) -> str:
+        print(f"### uh oh, abstract run vision")
```
Comment (aconchillo) [inline src/dailyai/services/ai_services.py:112]: remove print?

File: src/dailyai/services/ai_services.py [hunk 4]
```diff
@@ -109,12 +109,20 @@ def __init__(self, **kwargs):
...
         pass
 
     async def process_frame(self, frame: Frame) -> AsyncGenerator[Frame, None]:
+        print(f"### visionservice process frame, {frame}")
```
Comment (aconchillo) [inline src/dailyai/services/ai_services.py:116]: remove print?

File: src/dailyai/services/ai_services.py [hunk 5]
```diff
@@ -109,12 +109,20 @@ def __init__(self, **kwargs):
...
         if isinstance(frame, VisionImageFrame):
-            description = await self.run_vision(frame)
-            yield TextFrame(description)
+            print(f"### ### calling self.run_vision")
```
Comment (aconchillo) [inline src/dailyai/services/ai_services.py:118]: remove

File: src/dailyai/services/open_ai_services.py [hunk 6]
```diff
@@ -56,3 +64,68 @@ async def run_image_gen(self, prompt: str) -> tuple[str, bytes, tuple[int, int]]
...
+                    {
+                        "type": "image_url",
+                        "image_url": {
+                            "url": f"data:image/jpeg;base64,{base64_image}"
+                        },
+                    },
+                ],
+            }
+        ]
+        print(f"!!! messages: {messages}")
```
Comment (aconchillo) [inline src/dailyai/services/open_ai_services.py:119]: remove

File: src/dailyai/services/ai_services.py [hunk 7]
```diff
@@ -109,12 +109,20 @@ def __init__(self, **kwargs):
...
+            if getattr(self, "run_vision_async"):
+                async for frame in self.run_vision_async(frame):
+                    yield frame
+
+            else:
+                description = await self.run_vision(frame)
+                yield TextFrame(description)
```
Comment (aconchillo) [inline src/dailyai/services/ai_services.py:125]: shouldn't `run_vision` be async already?

Comment (chadbailey59) [reply src/dailyai/services/ai_services.py:121]: It is, but naming things is hard 😂 the way you implemented the `run_vision` method for moondream returns a string that you wrap in a text frame, but the the way the OpenAI module returns completions is an AsyncGenerator, so I needed another `run_vision` method. It's clunky but I'm out of ideas

Comment (aconchillo) [reply src/dailyai/services/ai_services.py:121]: I think we should make Moondream return AsyncGenerator :-). I'll fix that tomorrow.

**Comment by vipyne** [comment]
ok to close @chadbailey59 ?

**Comment by vipyne** [comment]
closing because this is stale

**Pull request closed without merging**